src/lib.rs

```diff
@@ -893,6 +897,39 @@ impl Octocrab {
         }
     }
 
+    #[tracing::instrument(
+        level="debug",
+        name="github api call",
+        skip(self, request_builder, attempt),
+        fields(err, url=field::Empty, service.name=field::Empty, http.method=field::Empty, http.url=field::Empty, http.status_code=field::Empty, http.resend_count=field::Empty)
+    )]
+    async fn send_call(
+        &self,
+        request_builder: reqwest::RequestBuilder,
+        attempt: u32,
+    ) -> Result<reqwest::Response, reqwest::Error> {
+        let span = tracing::Span::current();
+        let service = self.base_url.host_str().unwrap_or(GITHUB_SERVICE);
+        span.record("service.name", service);
+        if attempt > 1 {
+            span.record("http.resend_count", attempt);
+        }
+        let request = request_builder.build()?;
+        span.record("http.method", request.method().as_str());
+        span.record("http.url", request.url().as_str());
+        let result = self.client.execute(request).await;
+        match &result {
+            Ok(v) => {
+                span.record("http.status_code", v.status().as_u16());
+            }
+            Err(e) => {
+                let status = e.status().and_then(|s| Some(s.as_u16()));
+                span.record("http.status_code", status);
+            }
+        };
+        result
+    }
```

Comment:
Had to pull things out to a sep async function because `span` would've moved if I used `.instrument(span)`, which then prevents recording fields after the request.